server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -2627,7 +2627,7 @@ public boolean postStateTransitionEvent(final StateMachine2.Transition<State, Vi
                 s_logger.info("Schedule a router reboot task as router " + vo.getId() + " is powered-on out-of-band. we need to reboot to refresh network rules");
                 _rebootRouterExecutor.execute(new RebootTask(vo.getId()));
         } else {
-            if (isOutOfBandMigrated(opaque)) {
+            if (isOutOfBandMigrated(opaque) && (vo.getType() == VirtualMachine.Type.DomainRouter)) {
```

Comment:
Both if and else have the same check for DomainRouter, should we clean up the code?